tests/js/articles.spec.js

```diff
@@ -0,0 +1,178 @@
+import {
+  createNewPost,
+  enablePageDialogAccept,
+  getEditedPostContent,
+  insertBlock,
+  publishPost,
+} from '@wordpress/e2e-test-utils';
+
+import {
+  selectBlockByName,
+  openSidebarPanelWithTitle,
+  typeInInputWithLabel,
+  typeInInputWithPlaceholderLabel,
+  clickElementByText,
+  typeInDropdownWithLabel,
+  clearPreviousTextWithLabel,
+  clearPreviousTextWithPlaceholder,
+} from './e2e-tests-helpers';
+
+import 'expect-puppeteer';
+
+const ARTICLES_INVALID_URL_WARNING = 'The URL must start with "https://"';
+
+describe( 'Articles block', () => {
+  beforeAll( async () => {
+    // This helps in overriding Wordpress dialogs preventing actions.
+    await enablePageDialogAccept();
+  } );
+  beforeEach( async () => {
+    // Before running each test, go to the create post page.
+    await createNewPost({
+      postType: 'page',
+      title: 'Test Articles block',
+    } );
+
+    // Insert block by title.
+    await insertBlock( 'Articles' );
+  } );
+
+  // This is the first test, tests starts with it().
+  it ( 'is inserted into the Editor', async () => {
+    // Check if block was inserted
+    expect( await page.$( '[data-type="planet4-blocks/articles"]' ) ).not.toBeNull();
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  }, 50000 );
+
+  it ( 'check all articles inputs', async () => {
```

Comment:
Some semantic nit-picking: We should probably make all of these strings into sentences that start with it. Otherwise it will be confusing/ambiguous to read.

Reply:
Although in some cases (like here) it might get a bit contrived to find something fitting with "it".

Reply:
Maybe
```
it ( ' Should put what you enter in the editor into the right attributes')
```